t/24-worker-webui-connection.t

```diff
@@ -206,6 +206,132 @@ subtest 'attempt to register and send a command' => sub {
     ) or diag explain \@happended_events;
 };
 
+my $default_ua = $client->ua;
+
+subtest 'retry behavior' => sub {
+    # use fake Mojo::UserAgent and Mojo::Transaction
+    {
+        package Test::FakeTransaction;
+        use Mojo::Base -base;
+        has error => undef;
+        has res   => undef;
+    }
+    {
+        package Test::FakeUserAgent;
+        use Mojo::Base -base;
+        has fake_error  => undef;
+        has start_count => 0;
+        sub build_tx {
+            my ($self, @args) = @_;
+            return Test::FakeTransaction->new;
+        }
+        sub start {
+            my ($self, $tx, $callback) = @_;
+            $tx->error($self->fake_error);
+            $self->start_count($self->start_count + 1);
+            $callback->($self, $tx);
+        }
+    }
+    my $fake_error = {message => 'some timeout'};
+    my $fake_ua    = Test::FakeUserAgent->new(fake_error => $fake_error);
+    $client->ua($fake_ua);
+
+    # mock retry delay so tests don't take forever
+    my $web_ui_connection_mock                = Test::MockModule->new('OpenQA::Worker::WebUIConnection');
+    my $web_ui_supposed_to_be_considered_busy = 1;
+    $web_ui_connection_mock->mock(
+        _retry_delay => sub {
+            my ($self, $is_webui_busy) = @_;
+            is($is_webui_busy, $web_ui_supposed_to_be_considered_busy, 'retry delay applied accordingly');
+            return 0;
+        });
```

Comment:
I don't like mocks that don't test that the mocked method has actually been called.